### PR TITLE
refactor performance experiment server

### PR DESCRIPTION
**Pull request by WeiweiAtGit**
Main changes:
1. Saving results as JSON instead of HTML. This can ease results comparison in Project Performance Experiment (Perf-X) #1143.
2. Save flags. Perf-X will need this to rerun the lighthouse (with some extra configurations).
3. Save file and construct report page URL in server.js instead of bin.ts. This hopefully can make bin.ts more readable and reduce coupling.

### CODE REVIEW

File: lighthouse-cli/bin.ts [hunk 1]
```diff
@@ -262,20 +262,14 @@ function runLighthouse(url: string,
...
-      // If --view, generate report.html, host it, and open it in the default browser
+      // If --view, save and host this experiment, and open report.html in the default browser
       if (flags.view) {
-        return performanceXServer.startServer(0).then((port: number) => {
-          const filePath = `${performanceXServer.FOLDERS.REPORTS}/${filename}`;
-          return Printer.write(results, 'html', filePath).then(() => {
-            opn(`http://localhost:${port}/reports/${filename}`);
-          });
-        });
+        return performanceXServer.hostExperiment(flags, results).then(opn);
```
Comment (brendankenny) [inline lighthouse-cli/bin.ts]: I'll have to take a look in the morning, but love this already :)

Comment (WeiweiAtGit) [inline lighthouse-cli/bin.ts]: Thanks :)

**Comment by WeiweiAtGit** [comment]
This version does not work very well once I add some new functionalities to it. Will probably refactor again soon.

**Comment by WeiweiAtGit** [comment]
Will refactor again soon to support rerunning experiment

File: lighthouse-cli/performance-experiment/server.js [hunk 2]
```diff
@@ -18,94 +18,62 @@
...
-const server = http.createServer(requestHandler);
-server.on('error', err => log.error('PerformanceXServer', err.code, err));
+/**
+ * Start the server with an arbitrary port and open report page in the default browser.
+ * @param {!Object} lighthouseParams
+ * @param {!Object} results
+ * @return {!Promise<string>} Promise that resolves when server is closed
+ */
+let lhResults;
+function hostExperiment(lighthouseParams, results) {
```
Comment (paulirish) [inline lighthouse-cli/performance-experiment/server.js]: right now this seems more like `serveReport` or `serveAndOpenReport`.  Once experiments are in place, then I think we could rename whatever the entry method is called.

Comment (WeiweiAtGit) [inline lighthouse-cli/performance-experiment/server.js]: Thanks. Already renamed it to serveAndOpenReport.

**Pull request merged**